### DISCUSSION TITLE
New QC Tests to Add

### DISCUSSION THREAD

**Discussion by kenkehoe** [ARM-DOE/ACT#294] [Ideas]
This is to track what QC tests need to be added. I added all the numpy masked array options and two more, persistence and single variable comparison.

**Comment by kenkehoe** [comment]
Need to add a more than two variable comparison. I'm thinking of a three or more variable comparison where it can narrow down which variable is incorrect so a variable is not flagged when the other variable is the issue.

**Comment by kenkehoe** [comment]
Need to add some normal distribution tests. Maybe a Grubbs test, or Grubbs expanded like the DQ Office currently employes.

**Comment by kenkehoe** [comment]
Will need something along the lines of a statistical covariance.

**Comment by AdamTheisen** [comment]
Have you checked to see what python libraries are already out there?  Could we build what we need based on the scipy stats functions? https://docs.scipy.org/doc/scipy/reference/stats.html

**Comment by AdamTheisen** [comment]
One other thought is how we would want to implement these.  Not sure if adding as method or variable makes more sense.

**Comment by kenkehoe** [comment]
I prefer the method way as it allows for passing different keywords for each test. Different tests will need different inputs.

**Comment by AdamTheisen** [comment]
@kenkehoe not sure if you looked, but I did not add the FFT test to the same class as the others.  Since it was instrument specific, I thought it would be best to leave as a separate function that one has to call.  Sound good?

**Comment by kenkehoe** [comment]
Yeah that is fine @AdamTheisen.

**Comment by AdamTheisen** [comment]
@kenkehoe Converting this issue to the discussion board since it's more idea gathering.
